vyper/ir/compile_ir.py

```diff
@@ -967,7 +967,7 @@ def adjust_pc_maps(pc_maps, ofst):
     return ret
 
 
-def assembly_to_evm(assembly, pc_ofst=0, insert_vyper_signature=False):
+def assembly_to_evm(assembly, pc_ofst=0, insert_vyper_signature=False, disable_vyper_signature=False):
```

Comment:
insert_vyper_signature and disable_vyper_signature both being arguments is a little weird. it should be the callers responsibility to decide what the value here should be

Reply:
correct, i agree, this was a short term resolution to avoid refactoring due to the recursive pattern L1011